API/things-api/app.js

```diff
@@ -37,47 +28,39 @@ app.get('/checkout/:id/:person/:qty', function(req, res) {
 * /params   id
 *           person
 *           qty
-* /brief    Route to add item back into inventory 
+* /brief    Route to add item back into inventory
 *
 * /author   Luke
 ****************************************************/
 app.get('/checkin/:id/:person/:qty', function(req, res) {
-    
-    transaction(req.params.id, req.params.person, req.params.qty, function(err, tranRes) {
-        if (err) {
-            res.status(500);
-            res.send('Database Error');
-        } else {
-            res.status(200);
-            res.send(tranRes);
-        }
-    });
+    transaction(req.params.id, req.params.person, req.params.qty, errResultHandler, res);
 });
 
 
+
 /****************************************************
 * /path     N/A
-* /brief    Helper function for checkin route 
+* /brief    Helper function for checkin route
 *
 * /author   Luke
 ****************************************************/
-var transaction = function(id, person, qty, retFunc) {
+var transaction = function(id, person, qty, retFunc, res) {
```

Comment:
I see what you did now... This should work, I would be careful with callbacks going all over the place, it can make the code very hard to read.